api/src/services/accesscontrols.js

```diff
@@ -72,16 +80,39 @@ const grantsObject = {
     projects: {
       'read:own': ['*', '!users'], // cannot read associated users to the project
     },
+    datasets: {
+      'create:any': ['*'],
+      'read:own': ['*'],
+      'update:own': ['*'],
```

Comment:
User is allowed to create any datasets. They are only allowed to read and update their own datasets (the ones created by them or the ones assigned to them via project-associations).